src/bindings/swift/gen_swift.rs

```diff
@@ -0,0 +1,615 @@
+        case {{ loop.index }}: return .{{ value|decl_enum_variant_swift }}
+        {% endfor %}
+        default: throw InternalError.unexpectedEnumCase
+        }
+    }
+
+    func lower(into buf: Writer) {
+        self.toFFIValue().lower(into: buf)
+    }
+
+    func toFFIValue() -> UInt32 {
+        switch self {
+        {% for value in e.values() %}
+        case .{{ value|decl_enum_variant_swift }}: return {{ loop.index }}
+        {% endfor %}
+        }
+    }
+}
+{%- endfor -%}
+
+{%- for rec in ci.iter_record_definitions() %}
+public struct {{ rec.name() }}: Lowerable, Liftable, Serializable {
+    {%- for field in rec.fields() %}
+    let {{ field.name() }}: {{ field.type_()|decl_swift }}
+    {%- endfor %}
+
+    // Default memberwise initializers are never public by default, so we
+    // declare one manually.
+    public init(
+        {%- for field in rec.fields() %}
+        {{ field.name() }}: {{ field.type_()|decl_swift }}{% if loop.last %}{% else %},{% endif %}
+        {%- endfor %}
+    ) {
+        {%- for field in rec.fields() %}
+        self.{{ field.name() }} = {{ field.name() }}
+        {%- endfor %}
+    }
+
+    static func lift(from buf: Reader) throws -> {{ rec.name() }} {
+        return try {{ rec.name() }}(
+            {%- for field in rec.fields() %}
+            {{ field.name() }}: {{ "buf"|lift_from_swift(field.type_()) }}{% if loop.last %}{% else %},{% endif %}
+            {%- endfor %}
+        )
+    }
+
+    func lower(into buf: Writer) {
+        {%- for field in rec.fields() %}
+        {{ field.name() }}.lower(into: buf)
+        {%- endfor %}
+    }
+}
+{% endfor %}
+
+{% for func in ci.iter_function_definitions() %}
+{%- match func.return_type() -%}
+{%- when Some with (return_type) %}
+
+public func {{ func.name() }}(
+    // TODO: More considered handling of labels (don't emit them
+    // for single-argument functions; others?)
+    {%- for arg in func.arguments() %}
+    {{ arg.name() }}: {{ arg.type_()|decl_swift }}{% if loop.last %}{% else %},{% endif %}
+    {%- endfor %}
+) -> {{ return_type|decl_swift }} {
+    let _retval = {{ func.ffi_func().name() }}(
+        {%- for arg in func.arguments() %}
+        {{ arg.name()|lower_swift(arg.type_()) }}{% if loop.last %}{% else %},{% endif %}
+        {%- endfor %}
+    )
+    return try! {{ "_retval"|lift_swift(return_type) }}
+}
+
+{% when None -%}
+
+public func {{ func.name() }}(
+    {%- for arg in func.arguments() %}
+    {{ arg.name() }}: {{ arg.type_()|decl_swift }}{% if loop.last %}{% else %},{% endif %}
+    {%- endfor %}
+) {
+    {{ func.ffi_func().name() }}(
+        {%- for arg in func.arguments() %}
+        {{ arg.name()|lower_swift(arg.type_()) }}{% if loop.last %}{% else %},{% endif %}
+        {%- endfor %}
+    )
+}
+
+{%- endmatch %}
+{% endfor %}
+
+{% for obj in ci.iter_object_definitions() %}
+public class {{ obj.name() }} {
+    private let handle: UInt64
+
+    {%- for cons in obj.constructors() %}
+    public init(
```

Comment:
Hrm, interesting name-collision case here. What should we do if the Rust side declares a normal member method (or maybe a staticmethod?) named `init`?

Reply:
My vote: error/crash.

Reply:
Yeah, let's add `init` and `deinit` to our future reserved word list (#10) for Swift.